open-craft/xblock-group-project-v2
# Admin dashboard - "external" group state

**Description:** This PR adds second state for each stage at group level. New state is an "external" state - have the group received enough reviews from other students?

**Testing instructions:** 
1. Author and configure Group Project v2. Make sure to have some diversity - normal and TA graded activities, a couple of groups and so on. 
2. Generate review assignments for all non-TA graded activities.
3. Perform some reviews (but not all) in Peer Review stages, both as students in normal activities and as TA in TA-graded activities.
4. Log in to Apros as internal/company/McK admin
5. Go to Admin V2 dashboard in Group Work (MCKA ADMIN -> Group Work -> "Access Dashboard V2")
6. Select Program, Course and PRoject
7. Jump to activity details view: for each activity, click blue arrow in the bottom-right corner of any stage in that activity that have it - this will take you to activity details view.
8. Note that two statuses are displayed for each activity: first is "internal" status - have all the users in the group provided grades to other groups; second is "external" - have students in other groups provided reviews for this group.

External status is calculated as follows:
* If all assigned reviewers completed their reviews -> Completed
* If any of the assigned reviewers have started the review (but not necessarily completed it, though GPv2 interface does not allow partial reviews) -> Partially complete
* Otherwise (all reviewers relax on the beach instead of reviewing) -> Not started

What to test:
External status is actually accurate. Hint: try expanding other groups and checking individual student statuses - they show which groups the student is to grade.

Changed behaviors: Before the change, only "internal" status was shown at the group level. SInce that status makes little sense in case of TA review, for TA-graded stages it instead shown "have TAs provided grades for the group", which is now known the same as "external" status. Due to implementation, it also implied that status of individual student reviews was assigned from TA review, i.e. if TA have reviewed the group, students got their "Compete" statuses, despite doing nothing (or doing something) in the stage.

This override is now removed, so "internal" status and individual student review statuses are once again actual for TA-graded activities, which means they always show "Not started" in normal course of actions - students are not required to perform reviews in TA graded activity, but nothing prevents them from doing so and their reviews are counted toward other group's scores.
